packages/signature-controller/src/types.ts

```diff
@@ -109,6 +133,9 @@ type SignatureRequestBase = {
 
   /** Version of the signTypedData request. */
   version?: SignTypedDataVersion;
+
+  /** Response from message decoding api. */
+  decodedRequest?: DecodedRequestInfo;
```

Comment:
`decodedData` to de-couple from the API and encapsulate the error also?

Reply:
fixed